src/HistoricalSubscription.js

```diff
@@ -0,0 +1,113 @@
+import AbstractSubscription from './AbstractSubscription'
+
+export default class HistoricalSubscription extends AbstractSubscription {
+    constructor(streamId, streamPartition, callback, options, groupKeys, propagationTimeout, resendTimeout) {
+        super(streamId, streamPartition, callback, groupKeys, propagationTimeout, resendTimeout)
+        this.resendOptions = options || {}
+        this.realTimeMsgsQueue = []
+        if (!this.resendOptions) {
+            throw new Error('Resend options must be defined in a historical subscription.')
+        }
+
+        if (this.resendOptions.from != null && this.resendOptions.last != null) {
+            throw new Error(`Multiple resend options active! Please use only one: ${JSON.stringify(this.resendOptions)}`)
+        }
+
+        if (this.resendOptions.msgChainId != null && typeof this.resendOptions.publisherId === 'undefined') {
+            throw new Error('publisherId must be defined as well if msgChainId is defined.')
+        }
+
+        if (this.resendOptions.from == null && this.resendOptions.to != null) {
+            throw new Error('"from" must be defined as well if "to" is defined.')
+        }
+
+        /** * Message handlers ** */
+
+        this.on('unsubscribed', () => {
+            this._clearGaps()
+        })
+
+        this.on('disconnected', () => {
+            this.setState(AbstractSubscription.State.unsubscribed)
+            this._clearGaps()
+        })
+
+        this.on('error', () => {
+            this._clearGaps()
+        })
+    }
+
+    async handleBroadcastMessage(msg, verifyFn) {
```

Comment:
Ah I think understand know, since this may be part of `CombinedSubscription` we need to store the messages so when we switch to RealTimeSubscription we can re-play them.